### DISCUSSION TITLE
Unproject screen coordinates on terrrain - missing implementation?

### DISCUSSION THREAD

**Discussion by dennemark** [maplibre/maplibre-gl-js#4438] [Q&A]
Hi,
does map.unproject properly consider terrain?
I followed the code leading to the pointLocation function in transform ( https://github.com/maplibre/maplibre-gl-js/blob/c01e231defa326ef3559e669f937defaeaa752e9/src/geo/transform.ts#L581 ), it then calls coordinateLocation and applies terrain as argument, but actually coordinateLocation does not have a terrain arg. Am I missing something?
I feeld like clicking on the map and calling map.unproject, I get the lngLat on a plane, but not on the terrain.

Someone knows more about it? Is there another function? I could investigate it otherwise.
Best

**Comment by HarelM** [comment]
Can you write a more elaborate example? Maybe even a failing test?

**Reply by dennemark** [reply]
should i contribute the failing test as PR?

**Reply by HarelM** [reply]
Only if you plan to follow up, otherwise I'm afraid no one will pick it up...

**Reply by dennemark** [reply]
Ok after investigating it a bit loser I realised I had to look at pointCoordinate and not at coordinateLocation. So terrain is actually called. I am going to assume that the calculation is correct even though not tested.
